### ISSUE TITLE
CNI base plugins version detection broken since CNI 1.4.0 (broken upstream)

### ISSUE THREAD

**Issue by bfqrst** [hashicorp/nomad#20263]
### Nomad version
Nomad v1.7.6
BuildDate 2024-03-12T07:27:36Z
Revision 594fedbfbc4f0e532b65e8a69b28ff9403eb822e

### Operating system and Environment details
Ubuntu Jammy

### Issue

Clients detail page in the Nomad UI shows 1.0.0 across the board for CNI plugins.

### Reproduction steps
Check the respective UI page in the GUI of a v1.7.6 client.

#### Expected Result
Something like this on a v1.7.5 client tab:

![image](https://github.com/hashicorp/nomad/assets/7373518/f83d4746-95da-4e2e-9cbc-fc732dcfe89e)

#### Actual Result

![image](https://github.com/hashicorp/nomad/assets/7373518/469126b5-85f2-4769-ac53-4e3d20de5129)

**Comment by tgross** [comment]
Hi @bfqrst I happen to have been working on something related to this recently, so this caught my eye right away. I checked this myself and with an older version of CNI plugins the fingerprint works fine, even with Nomad 1.7.6. But on the _latest_ version, it looks like they're missing their version strings.

Ex. with 1.1.1

```
$ /opt/cni/bin/portmap --version
CNI portmap plugin v1.1.1
```

But with 1.4.1:

```
$ /opt/cni/bin/portmap --version
CNI portmap plugin version unknown
CNI protocol versions supported: 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.4.0, 1.0.0
```

So I think this is an upstream bug in the plugin builds? Can you check that on your cluster as well?

**Comment by tgross** [comment]
Ah, yeah I see this is reported upstream: https://github.com/containernetworking/plugins/issues/1019

**Comment by tgross** [comment]
Fortunately we won't have to revert https://github.com/hashicorp/nomad/pull/15473 because we only required 0.4.0 and these will detect as 1.0.0 because of the way the version gets fingerprinted.

**Comment by bfqrst** [comment]
Ah, damn, could've checked myself upstream...

THX @tgross

**Comment by tgross** [comment]
Ok we'll leave this open until it's fixed upstream just so folks who come looking for it see it. Not much we can do from Nomad's side unfortunately, as the data just isn't in the `--version` output for us to get.

**Comment by bfqrst** [comment]
Understood, makes sense.

**Comment by tgross** [comment]
Upstream has re-released 1.4.1 (without a new version, see https://github.com/containernetworking/plugins/issues/1038).

**Comment by bfqrst** [comment]
Confirming that v1.5.0 (the latest as of today) also shows fine in Nomad...
